HelloWorldService/src/main/java/com/gpnk/weather/WeatherModule.java

```diff
@@ -16,14 +19,18 @@ protected void config() {
     }
 
     /**
-     * Produces the dark sky secret key
+     * Produces the dark sky secret key.
+     * You can get your own key here: https://darksky.net/dev
      */
+    @Singleton
     @Provides
     @Named("darkSkySecretKey")
     public String getDarkSkySecretKey() {
-        // TODO: get this from config
-        // This is an old api key for an account I signed up for.  It is no longer valid.
-        // Get your own key here: https://darksky.net/dev
-        return "1b06b5c498df5d71efac20db9ea91fd6";
+
+        Config config = ConfigUtil.load();
+        Config hwConfig = config.getConfig("weather");
```

Comment:
What does the hw stand for in hwConfig?

Reply:
Good point! That got left from a copy-paste. Renamed this variable to `weatherConfig`.